### ISSUE TITLE
Add a way to include all prerelease versions in a range

### ISSUE THREAD

**Issue by acarl005** [semver/semver#871]
Currently, prerelease versions can only be opted-in to for specific patch versions, e.g. the range `^3.0.0-0` will allow `3.0.0-beta.1`, but not `3.1.0-beta.1`. I think this is the optimal behavior for what most users expect. However, there are some use cases for allowing prereleases for everything in the range. For example, see [this discussion](https://github.com/vitest-dev/vitest/issues/1960). In this case, vite-node wraps vite and it lists vite in its dependencies. However, @sveltejs/kit also depended upon vite, but it needed a prerelease beta version. The result is that @sveltejs/kit and vite-node ended up with separate, incompatible versions, breaking my project.

It would be nice if there was a way for vite-node to specify a looser range that allows all prerelease versions.

Maybe it could be something like `^^3.0.0-0`?

**Comment by ljharb** [comment]
Can’t it do `^3.0.0-0 || ^3.1.0-0` etc, as needed?

**Comment by acarl005** [comment]
Yes, that is possible. However, it means that the maintainers of vite-node need to add a version with every minor release of vite, which is burdensome.

**Comment by ljharb** [comment]
True, but that’s far less burdensome than updating the entire node ecosystem to support a new range operator.

**Comment by jwdonahue** [comment]
This seems like a discussion for the [node issues site](https://github.com/nodejs/node/issues).  SemVer, does not currently define range specs, though that is a [WIP](https://github.com/semver/semver/tree/isaacs/ranges), but it's primarily meant to document the most common factors between a small set of existing tools that each have their own range spec semantics.

**Comment by ljharb** [comment]
Oh, my mistake, I thought this issue was filed on npm/semver. This spec doesn't have ranges at all, so I totally agree it's entirely off topic.

**Comment by sabrican3532** [comment]
@segiddins

**Comment by anafebacus** [comment]
Tank u

**Comment by wojtekxtx** [comment]
Would not be sufficient to run sth along the lines:
```javascript
let rangeStart = Math.op().minimum(0);
let rangeStop = Math.op().maximum(255);
let minVer, maxVer = S(ret(0, scum));

minver = rangeStop - rangeStart;
```
?
